src/dedup.rs

```diff
@@ -116,32 +138,54 @@ async fn dedup_img(
     bot: &BotType,
     msg: &Message,
     db: impl Acquire<'_, Database = Postgres> + Send,
-) -> Result<Option<i32>, BoxedError> {
+) -> Result<Option<Duplicate>, BoxedError> {
     let chat_id = msg.chat.id.0;
     let message_id = msg.id.0;
 
-    if let Some(photo) = msg.image(bot).await? {
-        let hash = tokio::task::spawn_blocking(move || hash_img(&photo)).await?;
-        debug!(chat_id, message_id, hash, "image found");
+    let Some(photo) = msg.image(bot).await? else {
+        return Ok(None);
+    };
 
-        let mut conn = db.acquire().await?;
-        let record = sqlx::query_file!("sql/sim_img.sql", hash, chat_id)
-            .fetch_optional(&mut *conn)
-            .await?;
+    let hash = tokio::task::spawn_blocking(move || hash_img(&photo)).await?;
+    debug!(chat_id, message_id, hash, "image found");
 
-        if record.as_ref().map_or(true, |record| record.dist != 0) {
-            // A different image, need to insert the entity
-            sqlx::query_file!("sql/insert_img.sql", hash, chat_id, message_id)
-                .execute(&mut *conn)
-                .await?;
-        }
+    let mut conn = db.acquire().await?;
+    let record = sqlx::query_file!("sql/sim_img.sql", hash, chat_id)
+        .fetch_optional(&mut *conn)
+        .await?;
 
-        if let Some(record) = &record {
-            if !record.ignore {
-                return Ok(Some(record.message_id));
-            }
+    if let Some(record) = &record {
+        if !record.ignore {
+            let dup = Duplicate {
+                msg_id: record.message_id,
+                duplicate_cnt: record.duplicate_cnt,
+            };
+            return Ok(Some(dup));
         }
     }
 
+    if record.map_or(true, |record| record.dist != 0) {
+        // A different image, need to insert the entity
+        sqlx::query_file!("sql/insert_img.sql", hash, chat_id, message_id)
```

Comment:
`insert_img.sql` should be modified: `ON CONFLICT SET duplicate_count = ... + 1`